src/composed/message/parser.rs

```diff
@@ -8,13 +8,16 @@ use crate::errors::Result;
 use crate::packet::Packet;
 use crate::types::Tag;
 
-pub struct MessageParser<I: Sized + Iterator<Item = Packet>> {
+pub struct MessageParser<I: Sized + Iterator<Item = Result<Packet>>> {
     source: Peekable<I>,
 }
 
-fn next<I: Iterator<Item = Packet>>(packets: &mut Peekable<I>) -> Option<Result<Message>> {
+fn next<I: Iterator<Item = Result<Packet>>>(packets: &mut Peekable<I>) -> Option<Result<Message>> {
     while let Some(packet) = packets.by_ref().next() {
-        // for packet in packets.by_ref() {
+        let Ok(packet) = packet else {
+            return Some(Err(packet.expect_err("must be err")));
```

Comment:
rather than using `expect` just have a match statement

```rust
let packet = match packet {
  Ok(packet) => packet,
  Err(err) => return Some(Err(err)),
};
```